File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/module/TenantModuleRepository.java
package br.com.deltaglobalbank.identity.domain.module;

import java.util.List;
import java.util.UUID;

public interface TenantModuleRepository {
    List<TenantModule> findAllByTenantId(UUID tenantId);

    List<TenantModule> findAllByTenantIdAndEnabled(UUID tenantId, boolean enabled);

    TenantModule findByTenantIdAndModuleId(UUID tenantId, UUID moduleId);

    TenantModule save(TenantModule tenantModule);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/module/TenantModuleRepository.kt
package br.com.deltaglobalbank.identity.domain.module

import java.util.UUID

interface TenantModuleRepository {
    fun findAllByTenantId(tenantId: UUID): List<TenantModule>
    fun findAllByTenantIdAndEnabled(tenantId: UUID, enabled: Boolean): List<TenantModule>
    fun findByTenantIdAndModuleId(tenantId: UUID, moduleId: UUID): TenantModule?
    fun save(tenantModule: TenantModule): TenantModule
}
